queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/shares.py

```diff
@@ -71,6 +71,73 @@ def update_share_structure(business: Business, share_structure: Dict) -> Optiona
     return err
 
 
+def update_share_structure_correction(business: Business, share_structure: Dict) -> Optional[List]:
+    """Manage the share structure for a business.
+
+    Assumption: The structure has already been validated, upon submission.
+
+    Other errors are recorded and will be managed out of band.
+    """
+    if not business or not share_structure:
+        # if nothing is passed in, we don't care and it's not an error
+        return None
+
+    err = []
+
+    # Delete the ones that are present in db but not in the json and create the ones in json but not in db.
+    if resolution_dates := share_structure.get('resolutionDates'):
+        # Two lists of dates in datetime format
+        business_dates = [item.resolution_date for item in business.resolutions]
+        parsed_dates = [parse(resolution_dt).date() for resolution_dt in resolution_dates]
+
+        # Dates in both db and json
+        inclusion_dates = [business.resolutions[index] for index, date in enumerate(business_dates)
+                           if date in parsed_dates]
+        if len(inclusion_dates) > 0:
+            business.resolutions = inclusion_dates
+
+        # Dates in json and not in db
+        exclusion_dates = [date for date in parsed_dates if date not in business_dates]
+
+        resolution_dates = exclusion_dates if len(exclusion_dates) > 0 else parsed_dates
```

Comment:
won't `exclusion_dates` be good enough to use for appending new `resolutions` entries to the `business` object? I'm not seeing the need for L102...maybe I'm missing something.

Reply:
You are right. `resolution_dates = exclusion_dates` is enough for the looping below. I'll change that.